csharp/src/Google.Protobuf/WellKnownTypes/TimestampPartial.cs

```diff
@@ -115,9 +114,9 @@ public DateTime ToDateTime()
         {
             if (!IsNormalized(Seconds, Nanos))
             {
-                throw new InvalidOperationException(@"Timestamp contains invalid values: Seconds={Seconds}; Nanos={Nanos}");
+                throw new InvalidOperationException($"Timestamp contains invalid values: Seconds={Seconds}; Nanos={Nanos}");
             }
-            return UnixEpoch.AddSeconds(Seconds).AddTicks(Nanos / Duration.NanosecondsPerTick);
+            return new DateTime((Seconds + BclSecondsAtUnixEpoch) * TimeSpan.TicksPerSecond + (Nanos / Duration.NanosecondsPerTick), DateTimeKind.Utc);
```

Comment:
Note to self: this is always okay in terms of bounds, because we've checked for normalization.